dill/_dill.py

```diff
@@ -268,17 +276,52 @@ def dump(obj, file, protocol=None, byref=None, fmode=None, recurse=None, **kwds)
     return
 
 def dumps(obj, protocol=None, byref=None, fmode=None, recurse=None, **kwds):#, strictio=None):
-    """pickle an object to a string"""
+    """
+    Pickle an object to a string.
+
+    *protocol* is the pickler protocol, as defined for Python *pickle*.
+
+    If *byref=True*, then dill behaves a lot more like pickle as certain
+    objects (like modules) are pickled by reference as opposed to attempting
+    to pickle the object itself.
```

Comment:
Thanks for the example on module under byref=True, but I think this section may need some additional elaboration: 

How would byref=True affect the pickling of classes, functions?
What about classes and functions created in a nested scope?
What about dynamically created modules without a source file?

What is the definition of a 'ref? Does 'ref' here stand for a Python fully qualified name to an object such as ("package.module.Class.NestedClass.member")?

Would we expect an error when byref=True and the object cannot be constructed from the reference?